### ISSUE TITLE
Test whether a sentence is in the imperative mood

### ISSUE THREAD

**Issue by SRachamim** [spencermountain/compromise#822]
I wasn't aware of this library. It look awesome!

I was looking for a string predicate which can help me test if a given string is a valid sentence in the imperative mood. e.g.:

- `isImperative('Clean your room')` should be `true`
- `isImperative('Cleaned your room')` should be `false`
- `isImperative('Close the door')` should be `true`
- `isImperative('Closing the door')` should be`false`
- `isImperative('Take out the trash')` should be`true`
- `isImperative('Sweet new trash')` should be`false`

Is there any way I can achieve this with the current tools `compromise` offers? If not, then this is a feature request :)

**Comment by spencermountain** [comment]
hey @SRachamim this is a good question, and I think the answer is a bit muddy.

The pos-tagger in compromise assumes the input is correct, and does some dancing-around, to interpret things and make the sentence fit. Others have tried making a grammar-corrector in compromise, but I don't think anyone has been successful.
That said, compromise is not particularly *good*, or smart, and when it comes to complex verb forms, may just let dissonant verb-forms be - in which case your feature would be pretty-easy to do.

I guess my answer is an 'i don't know'. I'm curious how this goes - it's worth poking-around at.
Try inspecting different sentences with `.debug()` and see if you can find any patterns. You may be able to get somewhere with something like:
```js
const isImperative=function(doc){
  // 'clean the x'
  if(doc.has('#Infinitive (#Determiner|#Article|#Possessive) #Noun')){
    return true
  }
  //short sentence?
  if(doc.has('^#Infinitive .{1,2}$')){
   return true
  }
  return false
}
```
let me know how it goes!
cheers

**Comment by spencermountain** [comment]
haha - this rang a bell and I just did a search for 'imperative' - hopefully you can do better than [I did](https://github.com/spencermountain/compromise/blob/f1c5b50768f4a00768938d37f85a8866aafca20f/src/Subset/Verbs/conjugate/imperative.js) 😥

**Comment by spencermountain** [comment]
haha, i've gotten totally triggered by this question - am working on a method now.
Adding a `.isInfinitive()` method. should be within a day or two.

**Comment by spencermountain** [comment]
`.verbs().isImperative()` is released as `13.10.1` 
cheers

**Comment by SRachamim** [comment]
Wow, mate! Thanks for the fast response and reaction! You really made my day 🚀
